master/buildbot/secrets/providers/vault_hvac.py

```diff
@@ -0,0 +1,164 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+"""
+HVAC based providers
+"""
+
+
+from buildbot import config
+from buildbot.secrets.providers.base import SecretProviderBase
+
+import hvac
```

Comment:
hvac should be optional dependency.  For those, we usually do a try import and provide a more meaningfull message, e.g. what to pip install (so that the user don't guess and be vulnerable to typo squatting)

Reply:
I'm somehow stuck at making hvac optional dependency. While it's ok to make it optional in vault_hvac.py, I still need it for unit tests to simulate expired token (`raise hvac.exceptions.Unauthorized` in `FakeHvacKvV*`). If I understand it correctly, when I remove hvac from requirements-ci.txt, unit tests will fail in clean environment. Is there some way to specify dependency for unit tests only? Or any way how to overcome this?

Reply:
Having hvac in requirements-ci.txt is okay. What we are trying to achieve is that tests ran when hvac is not installed would just skip themselves. See e.g. how master/buildbot/test/unit/steps/test_http.py is setup.